scripts/babel-relay-plugin/src/RelayQLPrinter.js

```diff
@@ -553,6 +554,19 @@ function validateConnectionField(field: RelayQLField): void {
       field.getName()
     );
   });
+
+  if (!connectionNodeType.alwaysImplements('Node')) {
```

Comment:
Hmm... in practice, I think this will be easily missed when people develop their app. How about doing what we did with errors and printing a `console.warn` at runtime?

We can buffer all warnings on the `RelayQLPrinter` instance (yay, more reason to justify the fact that it is a class) and changing `RelayQLTransformer.transform` to return `{node: Printable, warnings: Array<string>}`?